docs/book/checkout.rst

```diff
@@ -0,0 +1,259 @@
+.. index::
+   single: Checkout
+
+Checkout
+========
+
+**Checkout** is a process that begins when the Customer decides to finish their shopping and transform their **Cart** into an **Order**.
+
+Checkout State Machine
+----------------------
+
+The Order Checkout state machine has 5 states available: ``cart``, ``addressed``, ``shipping_selected``, ``payment_selected``, ``completed``
+and a set of defined transitions between them - which are saved as the **checkoutState** of the **Order**.
+
+Besides the steps of checkout, each of them can be done once more. For instance if the Customer changes their mind
+and after selecting payment they want to change the shipping address they have already specified, they can of course go back and readdress it.
+
+The transitions on the order checkout state machine are:
+
+.. code-block:: yaml
+
+   transitions:
+      address:
+          from: [cart]
+          to: addressed
+      readdress:
+          from: [payment_selected, shipping_selected, addressed]
+          to: cart
+      select_shipping:
+          from: [addressed]
+          to: shipping_selected
+      reselect_shipping:
+          from: [payment_selected, shipping_selected]
+          to: addressed
+      select_payment:
+          from: [shipping_selected]
+          to: payment_selected
+      reselect_payment:
+          from: [payment_selected]
+          to: shipping_selected
+      complete:
+          from: [payment_selected]
+          to: completed
+
+Steps of Checkout
+-----------------
+
+Checkout in Sylius is divided into 4 steps. Each of these steps occurs when the Order goes into a certain state.
+See the Checkout state machine in the ``CoreBundle\Resources\config\app\state_machine.yml``
```

Comment:
It would be nice if instead of this path - namespaces we would give a direct link to file.